resampy/interpn.py

```diff
@@ -21,10 +19,9 @@ def resample_f(x, y, sample_ratio, interp_win, interp_delta, num_table):
 
     nwin = interp_win.shape[0]
     n_orig = x.shape[0]
-    n_out = y.shape[0]
     n_channels = y.shape[1]
 
-    for t in range(n_out):
+    for t, time_register in enumerate(t_out):
```

Comment:
I wonder - is there an implicit assumption that `t_out` is in sorted order here?  Does it fail if `t_out` is disordered, and/or do we need to have a check for this?

Reply:
IMHO t_out do not need to be sorted.
In the worst case I expect a very little slow down due to the higher probability to have CPU cache misses.

Reply:
Ok I could add something like the following to the resample_nu docstring:

```
x_out = resampy.resample_nu(x, t_new)
idx = np.arange(len(t_new))
np.random.shuffle(idx)
shuffled_x = resampy.resample_nu(x, t_new[idx])
np.testing.assert_array_equal(x_out[idx], shuffled_x)
```

... but probably it is better to have a dedicated unit test instead, IMHO.
@bmcfee please let me know what you prefer.

> There is an implied caveat that output y[n] corresponds to time t_out[n], which might be weird if t_out is unordered.

I agree with you in general, but there are some cases in which this is considered a feature, so I would not be incline e.g. in raising a warning if t_out is unordered.

Finally, according to your comment, I have implemented a parallel loop in `resample_f`.

Reply:
@bmcfee I have just realized that if I enable parallelism the tests fail on Python 3.5.
Unfortunately the problem seems to be related to an issue in numba so I'm not sure how to proceed now.
Should I switch off parallelism for the moment?

Reply:
@bmcfee I have just realized that if I enable parallelism the tests fail on Python 3.5.
Unfortunately the problem seems to be related to an issue in numba so I'm not sure how to proceed now.
Should I switch off parallelism for the moment?

Reply:
It's a good question - on the one hand, we don't really *need* parallel implementation here, so we could do this without bumping version requirements.  Have you compared speed of the parallel vs non-parallel implementations?  

On the other hand, 3.5 is over a year past [EOL](https://endoflife.date/python), so I don't think we need to continue to support it.  I doubt there are many people clamoring for non-uniform time sampling that are also pinned to python 3.5, and this is at least somewhat supported by our [download stats](https://pypistats.org/packages/resampy).

Reply:
The parallel version is 3.1 times faster that the sequential one on my 4 cores laptop (oversampling of a factor 4 on a an array of 110250 elements).

My suggestion is to switch off parallelism to be able to merge this PR ASAP.
Then we can eventually re-enable it and drop support to Python 3.5 in a separate PR.
Anyways, please consider that numba now supports Python >= 3.7 at the moment.

Personally I would prefer to have the parallel version enabled by default in the next release, but this is up to you to decide.
Maybe, we could add the capability to switch off parallelism using an environment variable, if necessary.

Reply:
> Anyways, please consider that numba now supports Python >= 3.7 at the moment.

That's true, but the question we should be asking is what the minimal numba version we can get away with is, not what the bleeding edge supports.

I generally agree that parallel should be enabled for this stuff, but you're right that it would be better handled in a separate PR just so we can take our time and get the dependencies worked out correctly.

Reply:
done